runtime/frequency/src/lib.rs

```diff
@@ -65,25 +66,30 @@ pub use common_runtime::{
 	weights,
 	weights::{BlockExecutionWeight, ExtrinsicBaseWeight},
 };
+use frame_support::traits::Contains;
 
 /// Basefilter to only allow specified transactions call to be executed
+/// For non mainnet [--features frequency] all transactions are allowed
 pub struct BaseCallFilter;
+
 impl Contains<Call> for BaseCallFilter {
-	fn contains(call: &Call) -> bool {
-		let core_calls = match call {
-			Call::System(..) => true,
-			Call::Timestamp(..) => true,
-			Call::ParachainSystem(..) => true,
-			Call::Sudo(..) => true,
-			Call::TechnicalCommittee(..) => true,
-			Call::Council(..) => true,
-			Call::Democracy(..) => true,
-			Call::Session(..) => true,
-			Call::Preimage(..) => true,
-			Call::Scheduler(..) => true,
-			_ => false,
-		};
-		core_calls
+	fn contains(_call: &Call) -> bool {
+		#[cfg(not(feature = "frequency"))]
+		{
+			true
+		}
+		#[cfg(feature = "frequency")]
+		{
+			matches!(
+				_call,
+				Call::System(..) |
+					Call::Timestamp(..) | Call::ParachainSystem(..) |
+					Call::Sudo(..) | Call::TechnicalCommittee(..) |
+					Call::Council(..) | Call::Democracy(..) |
+					Call::Session(..) | Call::Preimage(..) |
+					Call::Scheduler(..)
+			)
```

Comment:
when we build -all-features it will turn off not side of function scope and if we build with frequency flag it will activate the call filter side of scope

pretty much works like if-else but does not carry the non needed code to binary and wasm, hence its nice that if else but not opposed to having conditional separation rather


I have tested this though